supabase/migrations/limit_user_to_24_flashcard_sets.sql

```diff
@@ -0,0 +1,19 @@
+DROP FUNCTION IF EXISTS check_user_flashcard_sets_amount CASCADE;
+CREATE OR REPLACE FUNCTION check_user_flashcard_sets_amount() RETURNS TRIGGER AS $fn$
+  DECLARE
+    rc int;
+  BEGIN
+    rc := (SELECT count(*) FROM public.flashcard_sets WHERE user_id = NEW.user_id); 
+    RAISE NOTICE 'number of rows %', rc;
+    IF rc >= 24 THEN
+      RAISE EXCEPTION 'User has exceed maximum number of allowed flashcard sets. %', rc;
+    END IF;
+    RETURN NEW;
+  END;
+$fn$ LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS validate_flashcard_sets_limit_trigger on public.flashcard_sets CASCADE;
+CREATE TRIGGER validate_flashcard_sets_limit_trigger
+  BEFORE INSERT ON public.flashcard_sets
+  FOR EACH ROW
+  EXECUTE FUNCTION check_user_flashcard_sets_amount();
```

Comment:
Configure VSCode to insert newline at the end of file. I made a change here but not sure if it works for you https://github.com/kcjpop/vd/commit/61eb82773a9b928fb7d596b2a4812254ad61f9d5. If not, see https://stackoverflow.com/questions/44704968/visual-studio-code-insert-new-line-at-the-end-of-files